safemysql.class.php

```diff
@@ -201,12 +201,6 @@ public function transaction($callback, $can_retry = false)
 			}
 			$this->query('COMMIT');
 			return $result;
-		} catch (Throwable $e) {
-			$this->query('ROLLBACK');
-			throw $e;
-		} catch (Exception $e) {
-			$this->query('ROLLBACK');
-			throw $e;
```

Comment:
- [ ] You still need these! For one thing, you're no longer ever catching non-Exception throwables here, and for another, you now have no ROLLBACK ever occurring when `$can_retry = false`.